### DISCUSSION TITLE
Add `important` option to config

### DISCUSSION THREAD

**Discussion by mojtabamoosavi** [chakra-ui/panda#2949] [Q&A] [answered]
### Description

Add `important` option to control whether or not  utilities should be marked with `!important.`

### Problem Statement/Justification

I used Panda to build a design system, and now, for the documentation, I’m using a tool developed with Tailwind. However, the styles it generates are affecting the design system components. If there's a way to make all generated styles mark as `!important`, it could solve this issue.

### Proposed Solution or API

https://tailwindcss.com/docs/configuration#important

**Comment by aKus-dev** [comment]
I don't know if I understood you right, but I had a similar problem, I wanted to overwrite the styles of a component library, my solution was to add ``polyfill: true`` in panda config.

**Comment by cgatian** [comment]
Couldnt you control this with CSS specificity and layers?

**Comment by segunadebayo** [comment, marked as answer]
Thanks for the suggestion! After some thought, we're going to pass on this as a built-in config option. Here's our reasoning:

**`@layer` is the modern solution for this.** CSS cascade layers (`@layer`) give you explicit control over style priority without `!important`. Panda already generates all styles inside layers (`reset`, `base`, `tokens`, `recipes`, `utilities`), which means your styles will naturally take precedence when composed correctly with third-party CSS. Browser support is at 96%+ and climbing.

**`!important` on everything creates new problems.** Once all your styles are `!important`, you can't override them — compound variants fight each other, responsive overrides break, and consumer customization via the `css` prop becomes impossible.

**You can do this today with a plugin** if you really need it for a legacy embedding scenario:

```ts
import { defineConfig, definePlugin } from '@pandacss/dev'

const importantPlugin = definePlugin({
  name: 'important',
  hooks: {
    'cssgen:done': ({ content }) => {
      return content.replace(/;/g, ' !important;')
    },
  },
})

export default defineConfig({
  plugins: [importantPlugin],
  // ...
})
```

This keeps it opt-in and explicit rather than a global config flag. We'd recommend exploring `@layer` ordering or Shadow DOM encapsulation as more robust alternatives for third-party embedding.
